source/connections-survive-step-down/tests/README.rst

```diff
@@ -0,0 +1,224 @@
+===========================================
+Connections Survive Primary Step Down Tests
+===========================================
+
+These tests can be used to verify a driver's compliance with server discovery
+and monitoring requirements with respect to handling "not master" and
+"server shutting down" error responses from the server.
+
+.. contents::
+
+These tests apply only to replica set topologies.
+
+Tests
+-----
+
+The driver should implement the following tests:
+
+getMore Iteration
+`````````````````
+
+This test requires server version 4.2 or higher, and makes use of events
+defined in the `CMAP specification
+<https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst>`_.
+
+Perform the following operations:
+
+- Insert 5 documents into a collection.
+- Clear the connection pool to the current primary (see CMAP spec for
```

Comment:
Clearing the pool is a low level operation in a driver, and not something generally exposed via a MongoClient.

Consider requiring that the insert just above use a different MongoClient than the find below.  That way the pool can be made to be empty as the find is the first operation.

Reply:
I removed the language related to pool clearing, as it was more of an implementation detail than an essential part of the specification.